### PR TITLE
Auto infer version

### LINKED ISSUE #194: auto infer version (instead of expecting in exact format)
**Is your feature request related to a problem? Please describe.**
There's currently a long list of non-supported versions where we can easily infer the correct version. We could automatically infer the correct version to make yvm more user friendly

**Describe the solution you'd like**
Infer the correct version, and move a bunch of these invalid version cases, to the valid version cases:
https://github.com/tophat/yvm/blob/master/test/util/version.test.js#L38

**Describe alternatives you've considered**
Not doing it

**Additional context**

### PR DESCRIPTION
**Pull request by yashshah**
## Description

Add an option to infer range of version string which is prefixed with `v`. For example: `v1.0.0`  
Fix #194 

### DevQA Steps
`make install`
`yvm i v1.2.0`
`yvm use v1.2.0`

### Comments:

Right now regex is quite liberal with the text around the actual number, should we limit that only have optional prefix `v` before the version number?

### CODE REVIEW

**Comment by jakebolam** [comment]
@yashshah this looks great. CR looks good. Want to resolve merge conflicts and then I'll QA?

**Comment by jakebolam** [comment]
FYI valid version with this can be thing slike this too 🤔 `yvm use v1.9.4asdsadasda`

**Comment by yashshah** [comment]
You're right @jakebolam. Good catch! Something like `2.0.0-beta.3` can be valid version number. We also have `1.1.0-exp.2` in list-remote which we should also consider. I will make the change for this.

**Comment by yashshah** [comment]
@jakebolam I've fixed the conflicts. Can you please review this now?

File: src/yvm.js [hunk 1]
```diff
@@ -117,19 +120,22 @@ argParser
...
     .action(version => {
-        if (!isValidVersionString(version)) {
+        const parsedVersionString = getValidVersionString(version)
+        if (!parsedVersionString) {
             log('Error: Invalid version string supplied.')
             process.exit(1)
             return
         }
-        if (setDefaultVersion({version})) {
+        if (setDefaultVersion({ version })) {
```
Comment (jakebolam) [inline src/yvm.js]: Should this be the parsed version string?

**Comment by jakebolam** [comment]
@yashshah 1 comment. LGTM tho.

**Comment by yashshah** [comment]
@jakebolam Nice catch, I've made the changes. Can you please review again? Thanks!

File: src/yvm.js [hunk 2]
```diff
@@ -48,88 +47,99 @@ argParser
...
     .action(version => {
-        if (!isValidVersionString(version)) {
+        const parsedVersionString = getValidVersionString(version)
+        if (!parsedVersionString) {
             log('Error: Invalid version string supplied.')
             process.exit(1)
             return
         }
-        if (setDefaultVersion({version})) {
+        if (setDefaultVersion({ version: parsedVersionString })) {
```
Comment (jakebolam) [inline src/yvm.js]: @yashshah with named params we need to do this one like this :)

**Comment by jakebolam** [comment]
@yashshah feel free to merge this :)

**Pull request merged**